ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/CreateSourceFactory.java

```diff
@@ -142,14 +143,29 @@ public CreateStreamCommand createStreamCommand(
   }
 
   // This method is called by CREATE_AS statements
-  public CreateTableCommand createTableCommand(final KsqlStructuredDataOutputNode outputNode) {
+  public CreateTableCommand createTableCommand(
+      final KsqlStructuredDataOutputNode outputNode,
+      final Optional<RefinementInfo> emitStrategy
```

Comment:
Why is this being added here? Do we even support syntax that says that source tables should be created with EMIT FINAL?

Reply:
This is for `CTAS` statements, ie, persistent queries.